checker/src/main/java/alluxio/checker/HiveIntegrationChecker.java

```diff
@@ -0,0 +1,202 @@
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.PrintWriter;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+
+/**
+ * Some Hive queries to test the integration of Hive with Alluxio.
+ * Supports two options: use Alluxio as an option to store Hive tables
+ * and use Alluxio as the Hive default filesystem.
+ *
+ * This checker requires a running Hadoop cluster, but does not require a running Alluxio cluster.
+ * It will check whether Alluxio classes and Alluxio filesystem can be recognized
+ * in Hadoop task nodes.
+ */
+@SuppressFBWarnings("SQL_PREPARED_STATEMENT_GENERATED_FROM_NONCONSTANT_STRING")
+public class HiveIntegrationChecker {
+  private static final Logger LOG = LoggerFactory.getLogger(HiveIntegrationChecker.class);
+  private static final String FAIL_TO_CONNECT_TO_HIVE_MESSAGE = "Unable to connect to Hive, "
+      + "please check your Hive URL, username and password.\n";
+  private static final String FAIL_TO_FIND_CLASS_MESSAGE = "Please distribute "
+      + "the Alluxio client jar on the classpath of the application across different nodes.\n\n"
+      + "Please set HIVE_AUX_JARS_PATH either in shell or conf/hive-env.sh.\n"
+      + "For details, please refer to: "
+      + "https://www.alluxio.org/docs/master/en/Running-Hive-with-Alluxio.html\n";
+  private static final String FAIL_TO_FIND_FS_MESSAGE = "Please check the fs.alluxio.impl property "
+      + "in core-site.xml file of your Hadoop installation.\n\n"
+      + "For details, please refer to: "
+      + "https://www.alluxio.org/docs/master/en/Debugging-Guide.html\n";
+  private static final String TEST_FAILED_MESSAGE = "***** Integration test failed. *****\n";
+  private static final String TEST_PASSED_MESSAGE = "***** Integration test passed. *****\n";
+
+  @Parameter(names = {"-alluxiourl"}, description = "the alluxio cluster url of form "
+      + "alluxio://master_hostname:port")
+  private String mAlluxioURL = "";
+
+  @Parameter(names = {"-mode"}, description = "1 means use Alluxio as one option "
+      + "to store Hive tables, 2 means use Alluxio as the Hive default filesystem")
+  private int mUserMode = 1;
+
+  @Parameter(names = {"-hiveurl"}, description = "a database url "
+      + "of the form jdbc:subprotocol:subname")
+  private String mHiveURL = "";
+
+  @Parameter(names = {"-user"}, description = "the database user on whose behalf "
+      + "the connection is being made")
+  private String mHiveUserName = System.getProperty("user.name");
+
+  @Parameter(names = {"-password"}, description = "the user's password")
+  private String mHiveUserPassword = "";
+
+  /**
+   * Implements Hive with Alluxio integration checker.
+   *
+   * @param reportWriter save user-facing messages to a generated file
+   * @return 0 is success, 1 otherwise
```

Comment:
usually we either throw exception (see the signature) to indicate a failure code path, or return some error code but not throw exception in the signature. doing both see line 76 is not a good practice.
in here, I think we can return void and let the exception to speak for the error case.
In java, this is the typical practice --- exceptions are usually caught and turned into error code in the very top layer (e.g., main func)

Reply:
I think nearly all the exceptions in the run method has been caught by the  try-catch clause and nothing will be thrown through throw exception here. 

Maybe using a 0/1 error code is more meaningful here?

Reply:
SGTM